### PR TITLE
Added a weather command and added the API key to the /token command

### PR DESCRIPTION
**Pull request by rasmuselmgren**

### CODE REVIEW

File: src/commands/ts/weather.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    // Sends the current temperature and weather condition to the user.
+    return (
+      `The current temperature in *${city}* is: *${temp}*` +
+      '\u00B0C' +
+      ` and the current weather condition is *${weatherCondition}*`
+    );
+  } catch (error) {
+    // If the city is not found, sends an error message. Users cannot user Å, Ä and Ö.
+    return `Could not find weather data for *${city}* 😢 *Try again without using Å, Ä and Ö* `;
+  }
```
Comment (stefangeneralao) [inline src/commands/ts/weather.ts]: Will catch all errors. Failing API key, network errors and city not found will be caught. The return text will be misleading for some errors

File: src/commands/ts/token.ts [hunk 2]
```diff
@@ -20,6 +20,9 @@ const callback = (interaction: Interaction) => {
       interaction.user.send(
         `Execute the following command in the root of the project folder:\n\`\`\`echo "\\nDEV_TOKEN=${DEV_TOKEN} \\\n\\nGUILD_ID=${GUILD_ID} \\\n\\nDEV_CLIENT_ID=${DEV_CLIENT_ID}" >> .env\`\`\``
       );
+      interaction.user.send(`Optional keys you can add to your .env file:
+*WEATHER_API_KEY=587b112b73d3331d7bbfd7247516e478*
+      `);
```
Comment (stefangeneralao) [inline src/commands/ts/token.ts:25]: This doesn't share the format of the previous message. I suggest not having multiple responses (optional keys) as it forces the user to make multiple copy paste. I suggest exposing all environment variables from one message

File: src/commands/ts/weather.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    const weatherCondition = weather.data.weather[0].description;
+
+    // Sends the current temperature and weather condition to the user.
+    return (
+      `The current temperature in *${city}* is: *${temp}*` +
+      '\u00B0C' +
+      ` and the current weather condition is *${weatherCondition}*`
+    );
+  } catch (error) {
+    // If the city is not found, sends an error message. Users cannot user Å, Ä and Ö.
```
Comment (stefangeneralao) [inline src/commands/ts/weather.ts]: Personally not a fan of comments unless the code is hard to follow. The code is readable, therefore comments are not needed

File: src/commands/ts/weather.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    // Sets the temperature to a variable
+    const temp = weather.data.main.temp;
+    // Sets the weather condition to a variable
+    const weatherCondition = weather.data.weather[0].description;
+
+    // Sends the current temperature and weather condition to the user.
+    return (
+      `The current temperature in *${city}* is: *${temp}*` +
+      '\u00B0C' +
+      ` and the current weather condition is *${weatherCondition}*`
```
Comment (stefangeneralao) [inline src/commands/ts/weather.ts:30]: Why string concat?

File: src/commands/ts/weather.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+const name = 'weather';
+
+const description = 'Sends the weather for a given city';
+
+const callback = async (interaction: Interaction) => {
+  // Sets the city to the first argument
+  const city = interaction.options.get('city')!.value;
+  try {
+    // Gets the weather from the API
+    const weather = await axios.get(
```
Comment (stefangeneralao) [inline src/commands/ts/weather.ts:18]: Consider destructuring to `const { data } = await axios.get(`

File: src/commands/ts/weather.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
+import axios from 'axios';
+import { ApplicationCommandOptionType } from 'discord-api-types/v9';
+import { Interaction } from 'discord.js';
+import { IOption } from '../commands';
+
+// Imports API key from .env file
+const WEATHER_API_KEY = process.env.WEATHER_API_KEY;
```
Comment (stefangeneralao) [inline src/commands/ts/weather.ts:7]: Consider destructuring to `const { WEATHER_API_KEY } = process.env;`

**Review by stefangeneralao — CHANGES_REQUESTED** [review]

File: src/commands/ts/token.ts (hunk 2 above)
Comment (rasmuselmgren) [reply src/commands/ts/token.ts:25]: My thinking was that the user can chose which keys they want to use rather than getting all of them? Maybe it's better to have it all in one echo command even though some keys might not be used by the user

File: src/commands/ts/weather.ts (hunk 4 above)
Comment (rasmuselmgren) [reply src/commands/ts/weather.ts:30]: Because I'm not sure how to deal with \n stuff :D

File: src/commands/ts/token.ts (hunk 2 above)
Comment (stefangeneralao) [reply src/commands/ts/token.ts:25]: This is the way ✊

File: src/commands/ts/weather.ts (hunk 6 above)
Comment (rasmuselmgren) [reply src/commands/ts/weather.ts:7]: What's the purpose of this?

Comment (stefangeneralao) [reply src/commands/ts/weather.ts:7]: Scalability, readability, tidynessily and other buzzwords.
```
const WEATHER_API_KEY = process.env.WEATHER_API_KEY;
const BANANA_API_KEY = process.env.BANANA_API_KEY;
const APPLE_API_KEY = process.env.APPLE_API_KEY;
const PEAR_API_KEY = process.env.PEAR_API_KEY;
```
VS
```
const {
  WEATHER_API_KEY,
  BANANA_API_KEY,
  APPLE_API_KEY,
  PEAR_API_KEY
} = process.env;
```

File: src/commands/ts/weather.ts (hunk 4 above)
Comment (stefangeneralao) [reply src/commands/ts/weather.ts:30]: Try
```
`The current temperature in *${city}* is: *${temp}* \\u00B0C and the current weather condition is *${weatherCondition}*`
```

File: src/commands/ts/weather.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  if (!interaction.isCommand()) throw new Error('Not a command.');
+
+  const userInput = interaction.options.get('city')!.value as string;
+  const userInputToLowerCase = userInput.toLowerCase();
+  const city = userInputToLowerCase
+    .replace(/[\ö]/g, 'o')
+    .replace(/[\ñ]/g, 'n')
+    .replace(/[\ü]/g, 'u')
+    .replace(/[\å\ä\ã\á]/g, 'a')
+    .replace(/[^a-zA-Z0-9\ ]/g, '');
```
Comment (stefangeneralao) [inline src/commands/ts/weather.ts:22]: Hmmmm, smells a bit funky 😄

File: src/commands/ts/weather.ts [hunk 8]
```diff
@@ -9,6 +9,33 @@ const name = 'weather';
...
+const weatherConditionEmojiLabels: Record<WeatherCondition, string> = {
+  'clear sky': ':sunny:',
+  'few clouds': ':cloud:',
+  'scattered clouds': ':cloud:',
+  'broken clouds': ':cloud:',
+  'shower rain': ':cloud_rain:',
+  rain: ':cloud_rain:',
+  thunderstorm: ':cloud_lightning:',
+  snow: ':snowflake:',
+  mist: ':cloud:',
```
Comment (rasmuselmgren) [inline src/commands/ts/weather.ts:32]: Ska inte de vara strings? Eller ser jag fel?

**Review by stefangeneralao — APPROVED** [review]

File: src/commands/ts/weather.ts (hunk 8 above)
Comment (stefangeneralao) [reply src/commands/ts/weather.ts:32]: Nyckeln behöver inte vara en sträng om det ett ord med vanliga tecken. Med mellanslag eller dylikt behövs citattecken
